support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/authenticator/OAuth20ClientIdClientSecretAuthenticator.java

```diff
@@ -83,4 +101,32 @@ protected void validateCredentials(final UsernamePasswordCredentials credentials
             throw new CredentialsException("Bad secret for client identifier: " + credentials.getPassword());
         }
     }
+
+    /**
+     * Check if authentication can be performed for a given context.
+     *
+     * @param context the context
+     * @return true if authenticator can validate credentials.
+     */
+    protected boolean canAuthenticate(final WebContext context) {
+        val grantType = context.getRequestParameter(OAuth20Constants.GRANT_TYPE);
+
+        if (grantType.isPresent() && OAuth20Utils.isGrantType(grantType.get(), OAuth20GrantTypes.PASSWORD)) {
+            LOGGER.debug("Skipping Client credential authentication to use password authentication");
+            return false;
+        }
+
+        val code = context.getRequestParameter(OAuth20Constants.CODE);
+
+        if (!code.isEmpty()) {
+            LOGGER.debug("Checking if the OAuth code issued contains code challenge");
+            val token = this.ticketRegistry.getTicket(code.get(), OAuthCode.class);
+
+            if (token != null && token.getCodeChallenge() != null) {
+                LOGGER.debug("The OAuth code [{}] issued contains code challenge which requires PKCE Autentication", code.get());
```

Comment:
Typo; Should be `authentication` in lower case.

Reply:
Also, perhaps a @see tag would be good to add a reference to the PKCE authenticator, and in the javadoc for this method, you should include the problem description, etc too; as you have included in the PR here.

Reply:
👍